integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/SerializableMetadataIT.java

```diff
@@ -0,0 +1,105 @@
+    List<String> statements =
+        ImmutableList.of(
+            "CREATE KEYSPACE IF NOT EXISTS numbers WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'datacenter1' : 1 }",
+            "CREATE TABLE numbers.all (val int, doub int, str text, primary key (val,doub)) WITH CLUSTERING ORDER BY (doub DESC)",
+            "CREATE TYPE numbers.numbers_data (doub int, str text)",
+            "CREATE TABLE numbers.smaller (val int primary key, data numbers_data)",
+            "CREATE INDEX doub_idx on numbers.all (doub)",
+            "CREATE OR REPLACE FUNCTION numbers.avgState ( state tuple<int,bigint>, val int ) CALLED ON NULL INPUT RETURNS tuple<int,bigint> LANGUAGE java AS"
+                + " 'if (val !=null) { state.setInt(0, state.getInt(0)+1); state.setLong(1, state.getLong(1)+val.intValue()); } return state;'",
+            "CREATE OR REPLACE FUNCTION numbers.avgFinal ( state tuple<int,bigint> ) CALLED ON NULL INPUT RETURNS double LANGUAGE java AS"
+                + " 'double r = 0; if (state.getInt(0) == 0) return null; r = state.getLong(1); r/= state.getInt(0); return Double.valueOf(r);'",
+            "CREATE AGGREGATE IF NOT EXISTS numbers.average ( int ) SFUNC avgState STYPE tuple<int,bigint> FINALFUNC avgFinal INITCOND (0,0)",
+            "INSERT INTO numbers.all (val,doub,str) VALUES (1,2,'one')",
+            "INSERT INTO numbers.all (val,doub,str) VALUES (2,4,'two')",
+            "INSERT INTO numbers.all (val,doub,str) VALUES (3,6,'three')",
+            "CREATE MATERIALIZED VIEW IF NOT EXISTS numbers.big AS SELECT FROM numbers.all"
+                + " WHERE val is not null and doub is not null and val > 2"
+                + " PRIMARY KEY (val,doub)");
+
+    statements.stream().forEach((stmt) -> sessionRule.session().execute(stmt));
+  }
+
+  @Test
+  public void should_contain_serializable_metadata() {
+
+    Optional<KeyspaceMetadata> ksOption =
+        sessionRule.session().getMetadata().getKeyspace("numbers");
+    assertTrue(ksOption.isPresent());
+    KeyspaceMetadata ks = ksOption.get();
+    assertTrue(ks instanceof DefaultKeyspaceMetadata);
+
+    /* Validate that the keyspace metadata is fully populated */
+    assertFalse(ks.getUserDefinedTypes().isEmpty());
+    assertFalse(ks.getTables().isEmpty());
+    assertFalse(ks.getViews().isEmpty());
+    assertFalse(ks.getFunctions().isEmpty());
+    assertFalse(ks.getAggregates().isEmpty());
+
+    Optional<TableMetadata> tableOption = ks.getTable("all");
+    assertTrue(tableOption.isPresent());
+    TableMetadata table = tableOption.get();
+    assertTrue(table instanceof DefaultTableMetadata);
+
+    /* Validate that the table metadata is fully populated */
+    assertFalse(table.getPartitionKey().isEmpty());
+    assertFalse(table.getClusteringColumns().isEmpty());
+    assertFalse(table.getColumns().isEmpty());
+    // TODO: What kind of objects can be set for options here?  How do we guarantee they're
+    // serializable?
```

Comment:
Another odd case.  Options are presently defined as Objects and at the moment it looks like all of them are serializable (based on the contents of TableParser).  But this is hardly a guarantee going forward.  It'd be nice if we could change RelationMetadata.getOptions() to return a Map<CqlIdentifier, Serializable> but I can already hear @olim7t pointing out that that would be an API change (and with good reason).

Presumably we can just do the same thing here we did with the interfaces... instances of these interfaces are serializable as long as the concrete type is itself serializable.  In this case we'd have something in the docs saying the default table metadata impl is serializable as long as the options used are (which should be the common case).  Does that seem adequate?

Reply:
> I can already hear @olim7t pointing out that that would be an API change

I've thought it so loud that you heard me 😉 

+1 to add a note in the javadocs.

Note that this should remain mostly the responsibility of the driver; it's technically possible to plug custom schema logic that parses custom table options, but I don't think it's very likely to happen. So it will be up to us to keep that in mind and test it correctly if new option types get added.

Reply:
> I've thought it so loud that you heard me :wink: 

I mean, there's also the fact that I agree with you... the two work together nicely :)